src/dashboard/apps/auth/mixins.py

```diff
@@ -0,0 +1,15 @@
+"""Dashboard auth mixins."""
+
+from django.shortcuts import render
+
+
+class UserValidationMixin:
+    """Mixin to check if the request.user is validated."""
+
+    not_validated_template_name = "auth/user_not_validated.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        """Check if the user is validated."""
+        if not getattr(request.user, "is_validated", False):
+            return render(request, self.not_validated_template_name)
```

Comment:
Why not redirect to a dedicated view instead? I would find it more explicit, as a 403 view for example.